qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java

```diff
@@ -0,0 +1,214 @@
+                "ip.address": {
+                  "type": "ip"
+                }
+              }
+            }
+          }
+        }""";
+
+    public void testLogsIndexing() throws IOException {
+        if (isRunningAgainstOldCluster()) {
+            assertOK(client().performRequest(putTemplate(client(), "logs-template", STANDARD_TEMPLATE)));
+            assertOK(client().performRequest(createDataStream("logs-apache-production")));
+            final Response bulkIndexResponse = client().performRequest(bulkIndex("logs-apache-production", () -> {
+                final StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < randomIntBetween(10, 20); i++) {
+                    sb.append(
+                        String.format(
+                            BULK_INDEX_REQUEST,
+                            DateFormatter.forPattern(FormatNames.DATE_TIME.getName()).format(Instant.now()),
+                            randomFrom("foo", "bar"),
+                            randomFrom("PUT", "POST", "GET"),
+                            InetAddresses.toAddrString(randomIp(randomBoolean())),
+                            randomAlphaOfLengthBetween(100, 200)
+                        )
+                    );
+                    sb.append("\n");
+                }
+                return sb.toString();
+            }));
+            assertOK(bulkIndexResponse);
+            assertThat(entityAsMap(bulkIndexResponse).get("errors"), Matchers.is(false));
+        } else {
+            assertOK(client().performRequest(putTemplate(client(), "logs-template", LOGS_TEMPLATE)));
+            assertOK(client().performRequest(rolloverDataStream(client(), "logs-apache-production")));
+            final Response bulkIndexResponse = client().performRequest(bulkIndex("logs-apache-production", () -> {
+                final StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < randomIntBetween(10, 20); i++) {
+                    sb.append(
+                        String.format(
+                            BULK_INDEX_REQUEST,
+                            DateFormatter.forPattern(FormatNames.DATE_TIME.getName()).format(Instant.now()),
+                            randomFrom("foo", "bar"),
+                            randomFrom("PUT", "POST", "GET"),
+                            InetAddresses.toAddrString(randomIp(randomBoolean())),
+                            randomAlphaOfLengthBetween(100, 200)
+                        )
+                    );
+                    sb.append("\n");
+                }
+                return sb.toString();
+            }));
+            assertOK(bulkIndexResponse);
+            assertThat(entityAsMap(bulkIndexResponse).get("errors"), Matchers.is(false));
+
+            assertThat(
+                getSettings(client(), getWriteBackingIndex(client(), "logs-apache-production", 0)).get("index.mode"),
+                Matchers.nullValue()
+            );
+            assertThat(
```

Comment:
maybe for completeness also check whether synthetic source is enabled and index sorting is set? And also in the rolling upgrade test?

Reply:
The sorting if not explicitly applied is not part of the settings...I will test that in a different way as we discussed in Slack.